src/Menu.cpp

```diff
@@ -215,40 +215,97 @@ void Menu::DrawSettings()
 			ImGui::TableSetupColumn("##ListOfMenus", 0, 3);
 			ImGui::TableSetupColumn("##MenuConfig", 0, 7);
 
-			static size_t selectedMenu = SIZE_T_MAX;
+			static size_t selectedMenu = 0;
+
+			// some type erasure bs for virtual-free polymorphism
+			struct BuiltInMenu
```

Comment:
should this be defined inside function body? Doesn't cause any overhead?

Reply:
I feel like this is where they belong, for they are only used here, and the methods are easier to understand within context.

The visitor structs are essentially type switched drawing methods, encapsulated for std::visit. It could be written as an auto typed lambda with some constexpr type checks but i don't really like that.

Reply:
I don't think definition itself causes any overhead. The variants might introduce some, but should be less than using virtual functions and inheritance, and trivial as an UI function.